In [1]:
import pandas as pd

In [2]:
dataPath = "adidas.csv"

In [3]:
df = pd.read_csv(dataPath)

In [4]:
df.columns

Index(['Product Name', 'Product ID', 'Listing Price', 'Sale Price', 'Discount',
       'Brand', 'Description', 'Rating', 'Reviews', 'Last Visited'],
      dtype='object')

# 1. What are the total product counts for the Nike and Adidas brands?

In [5]:
#unique value shows multiple adidas brand, change them to Adidas.

brandCol = df['Brand'].unique()

In [6]:
#update column

for names in brandCol:
    if 'Adidas' in names:
        df['Brand'] = df['Brand'].str.replace(names, "Adidas")

In [7]:
#check for unique agian

df['Brand'].unique()

array(['Adidas', 'Nike'], dtype=object)

In [8]:
#count numbers
adidaCount = 0
NikeCount = 0
for brand in df['Brand']:
    if brand == "Adidas":
        adidaCount += 1
    else: 
        NikeCount += 1
        
#result
print (f"Adidas has total product counts of {adidaCount}, \nNike has total product counts of {NikeCount}")

Adidas has total product counts of 2625, 
Nike has total product counts of 643


## Adidas has total product counts of 2625. Nike has total product counts of 643

# 2. What is the average sales price for each brand?

In [9]:
#pull columns out and convert to list for easier useage
priceList = df['Sale Price'].tolist()
brandList = df['Brand'].tolist()

#check if is equal length
len(priceList) == len(brandList)

True

In [10]:
#overkill but fun

def averagePrice(brand: list, price: list):
    #declare variables
    adidasCount = 0
    nikeCount = 0
    adidasTotalPrice = 0
    nikeTotalPrice = 0
    
    #check if length is the same
    if (len(brand) != len(price)):
        print("lists must have same length to proceed")
        
    #update variables
    for i in range(len(brand)):
        if brandList[i] == 'Adidas':
            adidasCount += 1
            adidasTotalPrice += priceList[i]
        else:
            nikeCount += 1
            nikeTotalPrice += priceList[i]
            
    #calculate avg        
    adidasAvgPrice = adidasTotalPrice / adidasCount
    nikeAvgPrice = nikeTotalPrice / nikeCount
    
    #result
    print(f"The average price for adidas is ${adidasAvgPrice:.2f}\n,and the average price for nike is ${nikeAvgPrice:.2f}")

In [11]:
averagePrice(brandList, priceList)

The average price for adidas is $5135.00
,and the average price for nike is $10213.68


## The average price for adidas is \\$5135.00, and the average price for nike is \\$10213.68

# 3. What conclusions might you draw based on the answers to these questions above?

## Adidas offers more choice than Nike does, and the average price of the Adidas' is also lower than which of the Nike's. Adidas probably have more factores to make shoes and probably Adidas has higher market values.